packages/editor/src/components/post-excerpt/index.js

```diff
@@ -4,37 +4,88 @@
 import { __ } from '@wordpress/i18n';
 import { ExternalLink, TextareaControl } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
+import { useState } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import { store as editorStore } from '../../store';
 
-function PostExcerpt() {
-	const excerpt = useSelect(
-		( select ) => select( editorStore ).getEditedPostAttribute( 'excerpt' ),
+/**
+ * Renders an editable textarea for the post excerpt.
+ * Templates, template parts and patterns use the `excerpt` field as a description semantically.
+ * Additionally templates and template parts override the `excerpt` field as `description` in
+ * REST API. So this component handles proper labeling and updating the edited entity.
+ *
+ * @param {Object}  props                             - Component props.
+ * @param {boolean} [props.hideLabelFromVision=false] - Whether to visually hide the textarea's label.
+ * @param {boolean} [props.updateOnBlur=false]        - Whether to update the post on change or use local state and update on blur.
+ */
+export default function PostExcerpt( {
+	hideLabelFromVision = false,
+	updateOnBlur = false,
+} ) {
+	const { excerpt, shouldUseDescriptionLabel, usedAttribute } = useSelect(
+		( select ) => {
+			const { getCurrentPostType, getEditedPostAttribute } =
+				select( editorStore );
+			const postType = getCurrentPostType();
+			// This special case is unfortunate, but the REST API of wp_template and wp_template_part
+			// support the excerpt field throught the "description" field rather than "excerpt".
+			const _usedAttribute = [
+				'wp_template',
+				'wp_template_part',
+			].includes( postType )
+				? 'description'
+				: 'excerpt';
+			return {
+				excerpt: getEditedPostAttribute( _usedAttribute ),
+				// There are special cases where we want to label the excerpt as a description.
+				shouldUseDescriptionLabel: [
+					'wp_template',
+					'wp_template_part',
+					'wp_block',
+				].includes( postType ),
```

Comment:
Why there's wp_block here but not above in the "userAttribute" check. Should we use the same flag for both?

Reply:
I hate these inconsistencies, can we update the templates REST API to use excerpt and deprecate description?

Reply:
WDTY? Fine if tracked and done separately but I just feel like it doesn't really make sense for this to be different.

Reply:
You know at, I'm thinking more and doubting myself (I'm guessing pattern descriptions... and things like that also have description even if they're not post types). So I'm honestly not sure. Ok, let's try to forget about this inconsistency for now :P

Reply:
That would solve the `used attribute` issue, but it would not for:
>I think the main reason is that in the entities that we treat excerpt as a description feels better above in the card where we essentially describe the entity. In regular post types excerpt is more of part of the entity with a different functionality - describe the content.

Unless we completely split these two concepts and and add `description` to `wp_block` instead of the `excerpt` we recently added.

Reply:
yeah let's leave that for a separate discussion/PR. it's a rabbit hole that I'm not interested in digging now :)